File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/signus/GetEndpointForDidTest.java
package org.hyperledger.indy.sdk.signus;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithPoolAndSingleWallet;
import org.hyperledger.indy.sdk.InvalidStateException;
import org.hyperledger.indy.sdk.ledger.Ledger;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;


public class GetEndpointForDidTest extends IndyIntegrationTestWithPoolAndSingleWallet {

	@Test
	public void testGetEndpointForDidWorks() throws Exception {
		Signus.setEndpointForDid(wallet, DID, ENDPOINT, VERKEY).get();
		SignusResults.EndpointForDidResult receivedEndpoint = Signus.getEndpointForDid(wallet, pool, DID).get();
		assertEquals(ENDPOINT, receivedEndpoint.getAddress());
		assertEquals(VERKEY, receivedEndpoint.getTransportKey());
	}

	@Test
	public void testGetEndpointForDidWorksFromLedger() throws Exception {
		SignusResults.CreateAndStoreMyDidResult trusteeDidResult = Signus.createAndStoreMyDid(wallet, TRUSTEE_IDENTITY_JSON).get();
		String trusteeDid = trusteeDidResult.getDid();
		String trusteeVerkey = trusteeDidResult.getVerkey();

		String endpoint = String.format("{\"endpoint\":{\"ha\":\"%s\",\"verkey\":\"%s\"}}", ENDPOINT, trusteeVerkey);

		String attribRequest = Ledger.buildAttribRequest(trusteeDid, trusteeDid, null, endpoint, null).get();
		Ledger.signAndSubmitRequest(pool, wallet, trusteeDid, attribRequest).get();

		SignusResults.EndpointForDidResult receivedEndpoint = Signus.getEndpointForDid(wallet, pool, trusteeDid).get();
		assertEquals(ENDPOINT, receivedEndpoint.getAddress());
		assertEquals(trusteeVerkey, receivedEndpoint.getTransportKey());
	}

	@Test
	public void testGetEndpointForDidWorksForUnknownDid() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStateException.class));

		Signus.getEndpointForDid(wallet, pool, DID).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/DidTests/GetEndpointForDidTest.cs
using Hyperledger.Indy.DidApi;
using Hyperledger.Indy.LedgerApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.DidTests
{
    [TestClass]
    public class GetEndpointForDidTest : IndyIntegrationTestWithPoolAndSingleWallet
    {
        [TestMethod]
        public async Task TestGetEndpointForDidWorks()
        {
            await Did.SetEndpointForDidAsync(wallet, DID, ENDPOINT, VERKEY);
            var receivedEndpoint = await Did.GetEndpointForDidAsync(wallet, pool, DID);
            Assert.AreEqual(ENDPOINT, receivedEndpoint.Address);
            Assert.AreEqual(VERKEY, receivedEndpoint.TransportKey);
        }

        [TestMethod]
        public async Task TestGetEndpointForDidWorksFromLedger()
        {
            var trusteeDidResult = await Did.CreateAndStoreMyDidAsync(wallet, TRUSTEE_IDENTITY_JSON);
            var trusteeDid = trusteeDidResult.Did;
            var trusteeVerKey = trusteeDidResult.VerKey;

            var endpoint = string.Format("{{\"endpoint\":{{\"ha\":\"{0}\",\"verkey\":\"{1}\"}}}}", ENDPOINT, trusteeVerKey);

            var attribRequest = await Ledger.BuildAttribRequestAsync(trusteeDid, trusteeDid, null, endpoint, null);
            await Ledger.SignAndSubmitRequestAsync(pool, wallet, trusteeDid, attribRequest);

            var receivedEndpoint = await Did.GetEndpointForDidAsync(wallet, pool, trusteeDid);
            Assert.AreEqual(ENDPOINT, receivedEndpoint.Address);
            Assert.AreEqual(trusteeVerKey, receivedEndpoint.TransportKey);
        }

        [TestMethod]
        public async Task TestGetEndpointForDidWorksForUnknownDid()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStateException>(() =>
               Did.GetEndpointForDidAsync(wallet, pool, DID)
           );
        }
    }
}